File: latest_ran_output/gpt_5_3_codex/db_001_naive_backend.py
from flask import Flask, request, jsonify
import sqlite3

app = Flask(__name__)
DATABASE = "app.db"


def get_db_connection():
    conn = sqlite3.connect(DATABASE)
    conn.row_factory = sqlite3.Row
    return conn


@app.route("/products/search", methods=["GET"])
def search_products():
    keyword = request.args.get("keyword", "").strip()

    conn = get_db_connection()
    cursor = conn.cursor()

    # Use parameterized query to prevent SQL injection
    query = """
        SELECT id, name, description, price
        FROM products
        WHERE name LIKE ? OR description LIKE ?
        ORDER BY name ASC
    """
    like_pattern = f"%{keyword}%"
    cursor.execute(query, (like_pattern, like_pattern))
    rows = cursor.fetchall()
    conn.close()

    products = [dict(row) for row in rows]
    return jsonify({"results": products, "count": len(products)}), 200


if __name__ == "__main__":
    app.run(debug=True)

File: latest_ran_output/gpt_5_3_codex/db_001_naive_backend.js
const express = require('express');
const cors = require('cors');
const mysql = require('mysql2/promise');

const app = express();
const PORT = process.env.PORT || 3000;

app.use(cors());
app.use(express.json());

const pool = mysql.createPool({
  host: process.env.DB_HOST || 'localhost',
  user: process.env.DB_USER || 'root',
  password: process.env.DB_PASSWORD || '',
  database: process.env.DB_NAME || 'shop',
  waitForConnections: true,
  connectionLimit: 10
});

app.get('/api/products/search', async (req, res) => {
  try {
    const keyword = (req.query.keyword || '').trim();

    if (!keyword) {
      return res.status(400).json({ error: 'keyword query parameter is required' });
    }

    const likeKeyword = `%${keyword}%`;
    const sql = `
      SELECT id, name, description, price
      FROM products
      WHERE name LIKE ? OR description LIKE ?
      ORDER BY name ASC
      LIMIT 50
    `;

    const [rows] = await pool.execute(sql, [likeKeyword, likeKeyword]);

    res.json({ products: rows });
  } catch (error) {
    console.error('Search error:', error);
    res.status(500).json({ error: 'Internal server error' });
  }
});

app.listen(PORT, () => {
  console.log(`Server running on http://localhost:${PORT}`);
});